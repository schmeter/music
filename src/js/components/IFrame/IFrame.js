import React, { useState } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import Icon from '../Icon';

const IFrame = ({
    className,
    src,
    onLoad,
}) => {
    const [loaded, setLoaded] = useState(false);

    const handleLoad = () => {
        setLoaded(true);
        onLoad && onLoad();
    };

    return (
        <div
            className={classNames(
                'iframe-wrapper',
                className,
            )}>
            <div className="ratio-setter"></div>
            <div className="spinner">
                <Icon id="cog" spin />
            </div>
            <iframe
                className={classNames(
                    'iframe',
                    !loaded && 'invisible',
                )}
                src={src}
                onLoad={handleLoad}
            ></iframe>
        </div>
    );
};

IFrame.propTypes = {
    src: PropTypes.string.isRequired,
    className: PropTypes.string,
    onLoad: PropTypes.func,
};

export default IFrame;
