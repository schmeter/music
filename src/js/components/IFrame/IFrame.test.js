import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import IFrame from './IFrame';

describe('IFrame', () => {
    it('renders correctly', () => {
        const props = {
            src: 'test',
            className: 'test',
            onLoad: jest.fn(),
        };
        const component = mount(<IFrame {...props} />);

        component.simulate('load');

        expect(toJson(component)).toMatchSnapshot();
        // expect(props.onLoad).toHaveBeenCalled();
    });
});
