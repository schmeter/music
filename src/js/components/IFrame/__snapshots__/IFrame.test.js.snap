// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IFrame renders correctly 1`] = `
<IFrame
  className="test"
  onLoad={[MockFunction]}
  src="test"
>
  <div
    className="iframe-wrapper test"
  >
    <div
      className="ratio-setter"
    />
    <div
      className="spinner"
    >
      <Icon
        id="cog"
        spin={true}
      >
        <i
          className="fa-cog fa fa-spin"
        />
      </Icon>
    </div>
    <iframe
      className="iframe invisible"
      onLoad={[Function]}
      src="test"
    />
  </div>
</IFrame>
`;
